### ISSUE TITLE
Entstehungszeugnisse: Seitenzählung

### ISSUE THREAD

**Issue by gerritbruening** [faustedition/faust-gen-html#513]
@thvitt, vom Dienstleister bekomme ich (aufgeräumt) so was hier:
`<p>[18]</p>`
bedeutet
`<pb n="18"/>`
Jetzt suche ich nach einem Ausdruck [für das Skript hier](https://github.com/gerritbruening/TEI/blob/master/docx-cleanup.xsl), mit dem ich aus dem ersten das zweite machen kann.
Hast du da einen Rat?

**Comment by thvitt** [comment]
Probier mal:

```xslt
<xsl:template match="p[matches(., '\[\d+\]')]">
  <pb n="{replace(., '\[(\d+)\]', '$1')}"/>
</xsl:template>
```

**Comment by thvitt** [comment]
Setzt voraus, dass die Seitenzahlen _immer_ rein numerisch sind; für '1r' etc. ersetze die `\d` z.B. durch `\w` oder `[0-9rv]` oder so.

**Comment by gerritbruening** [comment]
Danke, das hilft bei dem gegebenen Fall sehr gut!
Allerdings kommt `\[d+\]` auch mitten in Absätzen vor, z.B.
```xml
<p><hi style="font-size:10pt">Herr Salomon Maimon, den er wegen ſeines großen
    Scharfſinns außerordentlich ſchätzte, und ich waren faſt die einzigen, die
    [(208)]er gern bei ſich ſah, weil er ſich unſertwegen nicht zu genieren
    brauchte.</hi>
```
Da erwischt dann auch `p/hi/text()[matches(., '\[\(\d+\)\]')]` zu viel Dokumentinhalt.

**Comment by thvitt** [comment]
Hast Du schon regeln für `text()`?

Sonst:

```xslt
<xsl:template match="text()">
  <xsl:analyze-string regex="\[\((\d+)\)\]">
    <xsl:matching-substring>
       <pb n="{regex-group(1)}"/>
    </xsl:matching-substring>
    <xsl:non-matching-substring>
        <xsl:value-of select="."/>
    </xsl:non-matching-substring>
  </xsl:analyze-string>
</xsl:template>
```

**Comment by gerritbruening** [comment]
OK:
https://github.com/faustedition/faust-gen-html/blob/master/tools/docx-cleanup.xsl#L37
Aber das hat nicht den Effekt, den ich erwartet hatte.
Beispieldaten liegen jetzt hier:
https://github.com/faustedition/xml/tree/master/testimony/docx2tei

**Comment by thvitt** [comment]
Zwei issues:

1) Ambiguous Rule Match → priority für die `text()`-Regel. Warnung sollte Dir Oxygen aber angezeigt haben.
2) Du hast die RE flasch abgeschrieben. Wenn die Seitenzahlen als `[(42)]` im Text liegen, ist der RE `\[\((\d+)\)\]`: `\[`/`\]` und `\(`/`\)` für die eckigen und runden Klammern, `(·)` um die innere Regular Expression (`\d+`) als `regex-group(1)` einzufangen

**Comment by gerritbruening** [comment]
> Warnung sollte Dir Oxygen aber angezeigt haben.

Ja, aber nur extrem kurz, und ich habe das nicht mehr aufrufen können. Ich habe dann geschaut, wo ich vielleicht doch noch auf `text()` matche, um die Ambiguität herauszufinden, aber ich habe keine Stelle gefunden.

Das mit den `(...)` war Absicht, denn manchmal steht auch `[42]` in den Daten.

**Comment by gerritbruening** [comment]
Schön wäre dann noch, wenn wir das darumherum liegende `p` samt `hi` beiseiteschaffen könnten, das jetzt ja nicht mehr gebraucht wird. Soll man das in einem nachgelagerten `xsl` machen und beide in ein `xproc` tun?

**Comment by thvitt** [comment]
![Fehlermeldungen der Transformation](https://user-images.githubusercontent.com/1906208/41304901-b0a61236-6e71-11e8-9a86-64432c44e5cf.png)

Man könnte ein XProc schreiben, solchen Kleinigkeiten tendiere ich aber eher dazu, das doch im Stylesheet mit Modi und Variablen fürs Zwischenergebnis zu machen.

**Comment by gerritbruening** [comment]
Ah, das war bei mir in einer langen Liste von Reitern ganz unten rechts versteckt, richtig.

**Comment by gerritbruening** [comment]
In Bezug auf das Ergebnis könnte man sagen: `p[not(node()[not(self::pb)])]` sollen mit `apply-templates` entfallen .

**Comment by gerritbruening** [comment]
Können sich @sandrakrause und @thvitt bei Gelegenheit anschauen.
